addons/wiki/migrations/0007_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,201 @@
+    logger.info('Creating new guids for all WikiPages:')
+    for i, wp in enumerate(WikiPage.objects.all()):
+        # looping instead of bulk_create, so _id's are not the same
+        progress_bar.update(i + 1)
+        Guid.objects.create(object_id=wp.id, content_type_id=content_type.id)
+    progress_bar.finish()
+    logger.info('WikiPage guids created.')
+    return
+
+def create_wiki_pages(state, nodes):
+    Guid = state.get_model('osf', 'guid')
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    NodeWikiPage = state.get_model('addons_wiki', 'nodewikipage')
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    wiki_pages = []
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    logger.info('Starting migration of WikiPages:')
+    for i, node in enumerate(nodes):
+        progress_bar.update(i + 1)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                node_wiki = NodeWikiPage.objects.get(id=Guid.objects.get(_id=version_list[0]).object_id)
+                latest_page_name = NodeWikiPage.objects.filter(id=Guid.objects.get(_id=version_list[-1]).object_id).values_list('page_name', flat=True).get()
+                wiki_pages.append(create_wiki_page(state, node, node_wiki, latest_page_name))
+    progress_bar.finish()
+    with disable_auto_now_add_fields(models=[WikiPage]):
+        WikiPage.objects.bulk_create(wiki_pages, batch_size=1000)
+    logger.info('WikiPages saved.')
+    create_guids(state)
+    return
+
+def create_wiki_versions(state, nodes):
+    NodeWikiPage = state.get_model('addons_wiki', 'nodewikipage')
+    WikiVersion = state.get_model('addons_wiki', 'wikiversion')
+    Guid = state.get_model('osf', 'guid')
+    wiki_versions_pending = []
+    guids_pending = []
+
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    logger.info('Starting migration of WikiVersions:')
+    for i, node in enumerate(nodes):
+        progress_bar.update(i + 1)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                node_wiki = NodeWikiPage.objects.get(id=Guid.objects.get(_id=version_list[0]).object_id)
+                page_name = NodeWikiPage.objects.filter(id=Guid.objects.get(_id=version_list[-1]).object_id).values_list('page_name', flat=True).get()
+                wiki_page = node.wikis.get(page_name__iexact=page_name.strip())
+                for index, version in enumerate(version_list):
+                    if index:
+                        node_wiki = NodeWikiPage.objects.get(id=Guid.objects.get(_id=version).object_id)
+                    wiki_versions_pending.append(create_wiki_version(state, node_wiki, wiki_page))
+                    guids_pending.append(migrate_guid_referent(state, Guid.objects.get(object_id=node_wiki.id, content_type=ContentType.objects.get_for_model(node_wiki).id)._id, wiki_page))
```

Comment:
This is querying the ContentType for `NodeWikiPage` for every version. We should only need to do this query once, since it doesn't change.